### PR TITLE
Upgrade to hardhat

### PR DESCRIPTION
**Pull request by zmanian**
Upgrade from Builder to hardhat

### CODE REVIEW

**Comment by taariq** [comment]
Very cool.

**Comment by jkilpatr** [comment]
this looks good to me, just let me update the contract deployer script and parsing that the tests require.

**Comment by zmanian** [comment]
Trying to make sense of this error
```
[2021-01-29T04:40:10Z INFO  test_runner::bootstrapping] stderr: (node:309) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/peggy/solidity/artifacts/TestERC20A.json'
        at Object.openSync (fs.js:476:3)
        at Object.readFileSync (fs.js:377:35)
        at getContractArtifacts (/peggy/solidity/contract-deployer.ts:165:41)
        at /peggy/solidity/contract-deployer.ts:95:31
        at step (/peggy/solidity/contract-deployer.ts:33:23)
        at Object.next (/peggy/solidity/contract-deployer.ts:14:53)
        at fulfilled (/peggy/solidity/contract-deployer.ts:5:58)
    (Use `node --trace-warnings ...` to show where the warning was created)
    (node:309) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
    (node:309) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
 ```

**Comment by zmanian** [comment]
It looks like the paths to artifacts changed.

**Review by jkilpatr — APPROVED** [review]
Looks good will merge after review from @jtremback

File: solidity/hardhat.config.ts [hunk 1]
```diff
@@ -19,12 +28,15 @@ usePlugin("buidler-typechain");
 // Go to https://buidler.dev/config/ to learn more
 module.exports = {
   // This is a sample solc configuration that specifies which version of solc to use
-  solc: {
+  solidity: {
     version: "0.6.6",
-    optimizer: { enabled: true, runs: 200000 }
```
Comment (jtremback) [inline solidity/hardhat.config.ts:24]: Might want to look into what different runs numbers do to the gas costs. IIRC the higher the better. Just costs more to deploy the contract

**Review by jtremback — COMMENTED** [review]
LGTM

**Pull request merged**